server/sockets/list.js

```diff
@@ -517,6 +518,50 @@ const removeListsOnArchiveCohort = (socket, dashboardClients) =>
       });
   });
 
+const emitListsOnRestoreCohort = (socket, dashboardClients, cohortClients) =>
+  socket.on(CohortActionTypes.RESTORE_SUCCESS, data => {
+    const { cohortId } = data;
+
+    List.find({ cohortId })
+      .populate('cohortId', 'ownerIds')
+      .lean()
+      .exec()
+      .then(docs => {
+        if (docs) {
+          const {
+            cohortId: { ownerIds: cohortOwners }
+          } = docs[0];
```

Comment:
maybe use `findOne`, so you don't have to use this [0] index notation.

Reply:
when restoring the cohort we have to send all cohorts' list meta-data to the clients, so we have to find all cohorts list and use `find`.
But there is a case that cohort's owners could be on archived cohort view so when the cohort is restoring they should get lists' meta-data also.  But we have to know cohort owners so I populate cohort data an get those owners from the first list (all list will have the same cohort data)

Reply:
That discussion only assures me there is something wrong with this design. Maybe you need a dedicated function to pull cohorts list metadata? The intent will be much cleaner then.